vsock-bridge/include/buffer.h

```diff
@@ -1,222 +1,63 @@
 #pragma once
 
-#include "logger.h"
-
-#include <algorithm>
+#include <array>
 #include <cassert>
 #include <cstdint>
-#include <iostream>
-#include <list>
-#include <memory>
-#include <vector>
-
-#include <unistd.h>
 
 namespace vsockio
 {
-	struct MemoryBlock
-	{
-		MemoryBlock(int size, class MemoryArena* region)
-			: _startPtr(std::make_unique<uint8_t[]>(size)), _region(region) {}
-
-		uint8_t* offset(int x) const
-		{
-			return _startPtr.get() + x;
-		}
-
-		std::unique_ptr<uint8_t[]> _startPtr;
-		class MemoryArena* _region;
-	};
-	
-	struct MemoryArena
-	{
-		std::vector<MemoryBlock> _blocks;
-		std::list<MemoryBlock*> _handles;
-		uint32_t _blockSizeInBytes = 0;
-		bool _initialized = false;
-
-		MemoryArena() = default;
-
-		void init(int blockSize, int numBlocks)
-		{
-			if (_initialized) throw;
-
-			Logger::instance->Log(Logger::INFO, "Thread-local memory arena init: blockSize=", blockSize, ", numBlocks=", numBlocks);
-
-			_blockSizeInBytes = blockSize;
-
-			for (int i = 0; i < numBlocks; i++)
-			{
-				_blocks.emplace_back(blockSize, this);
-			}
-
-			for (int i = 0; i < numBlocks; i++)
-			{
-				_handles.push_back(&_blocks[i]);
-			}
-			
-			_initialized = true;
-		}
-
-		MemoryBlock* get()
-		{
-			if (!_handles.empty())
-			{
-				auto mb = _handles.front();
-				_handles.pop_front();
-				return mb;
-			}
-			else
-			{
-				return new MemoryBlock(_blockSizeInBytes, nullptr);
-			}
-		}
-
-		void put(MemoryBlock* mb) 
-		{
-			if (mb->_region == this)
-			{
-				_handles.push_front(mb);
-			}
-			else if (mb->_region == nullptr)
-			{
-				delete mb;
-			}
-			else
-			{
-				throw;
-			}
-		}
-
-		int blockSize() const { return _blockSizeInBytes; }
-	};
-
 	struct Buffer
 	{
-		constexpr static int MAX_PAGES = 20;
-		int _pageCount;
-		int _cursor;
-		int _size;
-		int _pageSize;
-		MemoryBlock* _pages[MAX_PAGES];
-		MemoryArena* _arena;
-
-		explicit Buffer(MemoryArena* arena) : _arena(arena), _pageCount{ 0 }, _cursor{ 0 }, _size{ 0 }, _pageSize(arena->blockSize()) {}
-
-		Buffer(Buffer&& b) : _arena(b._arena), _pageCount(b._pageCount), _cursor(b._cursor), _size(b._size), _pageSize(b._arena->blockSize())
-		{
-			for (int i = 0; i < _pageCount; i++)
-			{
-				_pages[i] = b._pages[i];
-			}
-			b._pageCount = 0; // prevent _pages being destructed by old object
-		}
+        static constexpr int BUFFER_SIZE = 4096;
```

Comment:
In the description you mentioned buffer size 1024? Would it be useful to configure the buffer size for UID2 use cases?

Reply:
4096 is correct. This is the number we settled on based on perf testing. It is equal to the minimum socket send buffer size on Linux (added comment about that). This gives priority and performance for small requests/responses (e.g. token/generate or token/refresh) which tend to be real-time sensitive.